Change prototypes of extern functions of C++ plugins

rouault opened the pull request:
... to fix the -Wreturn-type-c-linkage warnings.

The methods that returned a QString now return a const QString*.

This is a breakage for out-of-tree native plugins.

I was considering changing the names of the methods, for example "namePtr" instead of "name", to detect more easily mismatches. I'm happy to do it if needed

3nids commented:
I am about to merge #43866 (everything is fixed now). This will probably mean to rebase your PR on top of it. If you see any issue, let me know.

rouault committed:
Fix build issue with clang 10.0 related to forward declaration of QgsFeatureRequest

rouault committed:
Change prototypes of extern functions of C++ plugins...

... to fix the -Wreturn-type-c-linkage warnings.

The methods that returned a QString now return a const QString*.

This is a breakage for out-of-tree native plugins.

rouault commented:
Rebased on top of latest master. Changes in auth providers dropped since no longer needed
